scripts/aptly/aptly_importer.py

```diff
@@ -0,0 +1,338 @@
+        self.__log_ok(f"aptly mirror {self.__get_mirror_name(distribution)} imported to the repo {repo_name}")
+
+    def __log(self, msg):
+        print(f" {msg} ")
+
+    def __log_ok(self, msg):
+        self.__log(f"   [ok] {msg}")
+
+    def __publish_new_snapshot(self, dist):
+        self.__log('Publish the new snapshot')
+        if (self.aptly.exists_publication(dist, self.__get_endpoint_name(dist))):
+            self.aptly.run(['publish', 'switch',
+                            dist,
+                            self.__get_endpoint_name(dist),
+                            self.__get_snapshot_name(dist)])
+            self.__log_ok(f"publish switch in {self.__get_endpoint_name(dist)} to use {self.__get_snapshot_name(dist)}")
+
+        else:
+            self.aptly.run(['publish', 'snapshot',
+                            f"-distribution={dist}",
+                            self.__get_snapshot_name(dist),
+                            self.__get_endpoint_name(dist)])
+            self.__log_ok(f"new publication in {self.__get_endpoint_name(dist)} for {self.__get_snapshot_name(dist)}")
+
+    def __remove_all_generated_mirrors(self):
+        for dist in self.config.suites:
+            mirror_name = self.__get_mirror_name(dist)
+            if self.aptly.exists(Aptly.ArtifactType.MIRROR, mirror_name):
+                self.aptly.run(['mirror', 'drop', mirror_name])
+
+    def run(self):
+        self.__log(f"\n == [ PROCESSING {self.config.name} ] ==\n")
+        # 1. Create aptly mirrors from yaml configuration file
+        # check aptly mirrors before creating to avoid problems beforehand
+        self.__assure_aptly_mirrors_do_not_exist()
+        for dist in self.config.suites:
+            self.__create_aptly_mirror(dist)
+            # 2. Be sure mirror has all source packages
+            self.__log(f'Check all source packages exist')
+            if not self.aptly.exists_all_source_packages(Aptly.ArtifactType.MIRROR,
+                                                         self.__get_mirror_name(dist)):
+                self.__remove_all_generated_mirrors()
+                self.__error(f'{self.__get_mirror_name(dist)} does not have a source package. Removing generated mirrors')
+            self.__log_ok('There is a source pakage in the mirror')
+            # 2. Import from mirrors to local repositories
+            self.__import__aptly_mirror_to_repo(dist)
+            # 3. Create snapshots from repositories
+            self.__create_aptly_snapshot(dist)
+            # 4. Publish new snapshots
+            self.__publish_new_snapshot(dist)
+
+        self.__log(f"\n == [ END OF PROCESSING {self.config.name} ] ==\n")
+        return True
+
+
+def main():
+    """
+    Usage: python3 aptly_importer.py <config_file>
+    """
+    usage = "usage: %prog config_file"
+    parser = argparse.ArgumentParser(usage)
+    parser.add_argument('config_file', type=str, nargs='+', default=None)
```

Comment:
Here nargs are set but only the first config_file is passed to the UpdaterManager. Is there a plan for handling more config files with multiple update managers or should we remove the nargs option (I'm fine with that to keep thinks simpler).

Reply:
good catch, that was a mistake. Removed in 0069bd7